modules/rke2/worker/instances_worker.tf

```diff
@@ -38,6 +38,53 @@ resource "aws_instance" "worker" {
   }
 }
 
+locals {
+        worker_with_eip = { for i, v in aws_instance.worker : tonumber(i)  => v.id if var.create_eip}      
+}
+
+resource "aws_eip" "worker_with_eip" {
+  depends_on = [var.dependency]
+  for_each = local.worker_with_eip
+  vpc = true
+  tags = {
+      Name ="${var.resource_name}-worker-${each.key}"
+    }
+}
+
+resource "aws_eip_association" "worker_eip_association" {
+        for_each = local.worker_with_eip
+        instance_id = aws_instance.worker[each.key].id
+        allocation_id = aws_eip.worker_with_eip[each.key].id
+        depends_on = [aws_instance.worker]
+} 
+
+resource "null_resource" "worker_eip" {
+
+  for_each = local.worker_with_eip
+  connection {
+    type = "ssh"
+    user = var.aws_user
+    host = aws_eip.worker_with_eip[each.key].public_ip
+    private_key = "${file(var.access_key)}"
+  }
+
+  provisioner "remote-exec" { 
+    inline = [
+      "sudo sed -i s/${local.master_ip_prev}/${local.master_ip}/g /etc/rancher/rke2/config.yaml",
+      "sudo sed -i s/-ip:.*/\"-ip: ${aws_eip.worker_with_eip[each.key].public_ip}\"/g /etc/rancher/rke2/config.yaml",
+      "sudo systemctl restart rke2-agent"
+    ]
+  }
+
+provisioner "local-exec" {
+    command    = "sed s/127.0.0.1/\"${local.master_ip}\"/g /tmp/${var.resource_name}_config >/tmp/${var.resource_name}_kubeconfig"
+    on_failure = continue
+  }
```

Comment:
Does this work on the worker nodes? I don't think they should have any kubeconfig.